neurots/generate/section.py

```diff
@@ -91,15 +96,43 @@ def update_pathlength(self, length):
         """Increases the path distance."""
         self.pathlength += length
 
-    def next_point(self, current_point):
-        """Returns the next point depending on the growth method and the previous point."""
+    def _propose(self, noise=0.0):
+        """Propose the direction for a next section point.
+
+        Args:
+            noise (float): artificially increase the randomness to allow for context changes
+        """
         direction = (
             self.params.targeting * self.direction
-            + self.params.randomness * get_random_point(random_generator=self._rng)
+            + self.params.randomness * get_random_point(random_generator=self._rng) * (1.0 + noise)
             + self.params.history * self.history()
         )
 
-        direction = direction / vectorial_norm(direction)
+        return direction / vectorial_norm(direction)
+
+    def next_point(self, current_point):
+        """Returns the next point depending on the growth method and the previous point.
+
+        If a context is present, an accept-reject mechanisms will be used to alter the next point.
+        """
+        if self.context is not None and "section_prob" in self.context:  # pragma: no cover
+            direction = accept_reject(
+                self._propose,
+                self.context["section_prob"],
+                self._rng,
+                max_tries=self.context.get("params_section", {}).get("max_tries", DEFAULT_MAX_TRIES)
+                if self.context is not None
```

Comment:
We already know that `self.context` is not None here since we tested it at line 118

Reply:
good catch, I simplified things and forgot that one